miner/worker.go

```diff
@@ -414,6 +473,15 @@ func (w *worker) newWorkLoop(recommit time.Duration) {
 					timer.Reset(recommit)
 					continue
 				}
+				//reset txpoolsnapshot and interrupt current pre-mining
```

Comment:
why reset here? 
`timer` if for resubmit for new arrived high priced transactions, the default resubmit interval is 3 second?
I am not sure, how this timer is used?